pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -995,7 +996,8 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) e
 }
 
 // AppendInstanceHandler implements a service catalog operation
-func (c *Controller) AppendInstanceHandler(func(*model.ServiceInstance, model.Event)) error {
+func (c *Controller) AppendInstanceHandler(f func(*model.ServiceInstance, model.Event)) error {
+	c.instanceHandlers = append(c.instanceHandlers, f)
```

Comment:
I think this is going to cause redundant full pushes. See https://github.com/istio/istio/blob/c764e1d51572e8b588ec9cf8f69fc2ce7e763cb1/pilot/pkg/bootstrap/server.go#L729. Previously we ignored these, now we will not only get the service entry handler, but also trigger a full push

Reply:
We really need a test for this... You are right, and there is no flag.

Reply:
Oh thats a good catch. Its an easy fix.. holdon